doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+
+
+
+
+# 4 Feature Design
+
+
+
+## 4.1 Design Overview
+
+### 4.1.1 Basic Approach
+
+For control plane, MP-BGP with EVPN extension support is leveraged from FRR with changes in Zebra and Fpmsyncd to interact with SONiC SwSS. A new process Fdbsyncd is added in SwSS to sync MAC route with kernel.  
+
+VXLAN orchagent and manager, FDB orchagent, VRF orchagent and manager, route orchagent   are enhanced to achieve this feature. 
+
+
+### 4.1.2 Container
+
+No new container is added. The changes are added to existing containers such as FRR, SwSS and Syncd. The details of the changes will be discussed in the Design section below.
+
+### 4.1.3 SAI Overview
+
+To support this feature, SAI will be extended as below:
+
+- **sai_fdb_entry_type_t** - will be enhanced to have a new enum. 
+  - SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE - This will be used when a MAC should not be aged out but a MAC move should be allowed. An example is remote MAC learnt by EVPN procedures which can be subjected to a MAC move from remote to local.
+- **sai_tunnel_attr_t** - It will be enhanced to have one new attribute
+  - SAI_TUNNEL_ATTR_ENCAP_DEST_IP - New tunnel attribute will be added to specify the tunnel destination IP address. This will be used to support warm reboot. Currently only the SIP is part of the tunnel attributes which does not allow for unique set of attributes when there are multiple tunnels with same SIP and different DIPs. The unique set of attributes is required for reconcilation during a warm reboot.  bridgeport associated with a tunnel also requires the tunnel id to be reconcilable post warm restart. Hence, it is required as part of the sai_tunnel_attr_t list.
+- **sai_vlan_attr_t** - It will be enhanced to have one new attribute
+  - SAI_VLAN_ATTR_NEIGH_SUPP_MODE - By default this value will be FALSE. Reason for this new attribute is due to recent changes in ARP behaviour. The default arp behavior in copp file is changed from trap to copy(as part of commit -5e4b71d4). But for ARP Suppression, we need Pkt to get trapped instead of copy to avoid flooding of packets over vxlan tunnel.So a differentiated tcam rule for these arp suppressed vlans will redirect the ARP pkts to cpu instead of copy.
+
+The changes are subject to approval by the SAI community.
+
+## 4.2 DB Changes
+
+At a high level below are some of the interactions between relevant components and the DB involved for EVPN L2 VXLAN support in SONiC architecture.
+
+![EVPN L2 VXLAN Arch](images/arch.PNG "Figure : Arch")
+__Figure 4: EVPN L2 VXLAN DBs__
+
+### 4.2.1 CONFIG_DB changes
+
+**EVPN_NVO_TABLE**
+
+Producer:  config manager 
+
+Consumer: VxlanMgr
+
+Description: This is a singleton configuration which holds the VTEP source IP used for BGP-EVPN discovered tunnels. 
+
+Schema:
+
+```
+;New table
+;holds the VTEP source IP used for BGP-EVPN discovered tunnels
```

Comment:
Is the description misleading here? We dont store the VTEP source IP here.. rather the Name of the Tunnel, which has the IP associated with.

Reply:
Will correct.